reference-target-explainer.md

```diff
@@ -0,0 +1,611 @@
+
+The equivalent declarative attribute is `shadowrootreferencetargetmap`, which is a comma-separated list of attribute to ID mappings.
+
+> Note: the syntax of `shadowrootreferencetargetmap` is based on the [`exportparts`](https://drafts.csswg.org/css-shadow-parts/#exportparts-attr) attribute that contains a comma-separated map of part names.
+
+```html
+<input
+  role="combobox"
+  aria-controls="fancy-listbox"
+  aria-activedescendant="fancy-listbox"
+/>
+<fancy-listbox id="fancy-listbox">
+  <template
+    shadowrootmode="closed"
+    shadowrootreferencetargetmap="aria-controls: real-listbox,
+                                  aria-activedescendant: option-1"
+  >
+    <div id="real-listbox" role="listbox">
+      <div id="option-1" role="option">Option 1</div>
+      <div id="option-2" role="option">Option 2</div>
+    </div>
+  </template>
+</fancy-listbox>
+```
+
+The JavaScript API reflects the mappings using camelCase names for the properties, and `htmlFor` for `for`:
+
+```js
+this.shadowRoot_.referenceTargetMap.ariaControls = "real-listbox";
+this.shadowRoot_.referenceTargetMap.ariaActiveDescendant = "option-1";
+this.shadowRoot_.referenceTargetMap.htmlFor = "real-input";
+```
+
+#### Live references
+
+Reference targets are a "live reference": if the host internally changes its reference target mapping, any element that references the host will use the updated mapping.
+
+In the example above, if the `aria-activedescendant` mapping is changed, then the `aria-activedescendant` of `<input>` will be changed to refer to the newly-mapped element.
+
+- **Before**: `<input aria-activedescendant="fancy-listbox">` initially maps to 'option-1'.
+- fancy-listbox internally updates its mapping:
+  ```js
+  this.shadowRoot_.referenceTargetMap.ariaActiveDescendant = "option-2";
+  ```
+- **After**: `<input aria-activedescendant="fancy-listbox">` now maps to 'option-2', without needing to update the input element itself.
+
+#### Combining `referenceTarget` and `referenceTargetMap`
+
+In the case where both attributes are specified, `referenceTargetMap` takes priority for individual attributes, and `referenceTarget` acts as the fallback for attributes that are not specified.
+
+In the example below, `"real-listbox"` is the target for all attributes _except_ `aria-activedescendant`, which is targeted to `"option-2"`.
+
+```html
+<input
+  role="combobox"
+  aria-controls="fancy-listbox"
+  aria-activedescendant="fancy-listbox"
+/>
+<fancy-listbox id="fancy-listbox">
+  <template
+    shadowrootmode="open"
+    shadowrootreferencetarget="real-listbox"
+    shadowrootreferencetargetmap="aria-activedescendant: option-2"
+  >
+    <div id="real-listbox" role="listbox">
+      <div id="option-1" role="option">Option 1</div>
+      <div id="option-2" role="option">Option 2</div>
+    </div>
+  </template>
+</fancy-listbox>
+```
```

Comment:
This is very powerful. This would support both the usecase where 1) the component author wants to explicitly limit the `aria-*` attributes that can be used to target specific elements, as well as 2) the case where the component author doesn't care and wants to just have any wildcard `aria-*` attribute point at a specific element. It also does so in such a way that doesn't force usecase #<span></span>2 down the explicit path if they need to set aside one `aria-*` attribute for a particular target (as in your `aria-activedescendant` example). I like it!